core/src/main/java/org/mskcc/cbio/portal/dao/DaoPatient.java

```diff
@@ -60,20 +83,20 @@ private static void clearCache()
     public static synchronized void reCache()
     {
         clearCache();
-
+        Vector<PrimitivePatientRecord> patientRecordsRetrieved = new Vector<>();
```

Comment:
just curious.. why a vector?

Reply:
The Vector class is thread safe, and we use synchronized methods in DaoCancerStudy .. presumably to prevent two parallel thread from both calling reCacheAll.
So if multithreading is a worry, it seemed safer to stay with the threadsafe class (rather than ArrayList).

Reply:
@sheridancbio maybe you can add a comment saying so? that's useful to know. either in the pr or here at this line?

Reply:
I will add comment to the code

Reply:
Actually, on deeper reflection, this variable is already inside a synchronized function call. And as a local variable, it would only have visibility to a single thread anyway. To say that Vector is "thread safe" really means that the read/write methods are synchronized on the object ... but we don't need that inside a synchronized method really. So I'm going to switch it to ArrayList